### PR TITLE
Bug fixes for threat intel

### PR DESCRIPTION
**Pull request by jowg-amazon**
### Description
This PR contains bug fixes that were found for threat intel:

- Fixes the initial creation for the default source config so that it catches an index not found exception.
- fixes the number of conditions created from a match query
- Fixes threat intel source update when trying to deactivate and activate
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/threatIntel/service/DefaultTifSourceConfigLoaderService.java [hunk 1]
```diff
@@ -64,8 +68,12 @@ public void createDefaultTifConfigsIfNotExists(ActionListener<Void> listener) {
                 ActionListener.wrap(searchResponse -> {
                     createTifConfigsThatDontExist(searchResponse, tifMetadataList, listener);
                 }, e -> {
-                    log.error("Failed to search tif config index for default tif configs", e);
-                    listener.onFailure(e);
+                    if (e instanceof IndexNotFoundException) {
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/threatIntel/service/DefaultTifSourceConfigLoaderService.java:71]: can you cover scenario fi cause is wrapped in a remote transport exception

Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/threatIntel/service/DefaultTifSourceConfigLoaderService.java:71]: can you add to upload ioc kinda test that calls search API for tif configs before creating the config and verify this behavior is fixed

**Comment by eirsep** [comment]
plz  add debug log when lock is released

plz open issue to make release lock a event driven call to wait upon before returning response

**Comment by jowg-amazon** [comment]
> plz add debug log when lock is released
> 
> plz open issue to make release lock a event driven call to wait upon before returning response

Created a new issue:
https://github.com/opensearch-project/security-analytics/issues/1224

**Review by AWSHurneyt — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Review by eirsep — APPROVED** [review]

**Pull request merged**